### PR TITLE
Don't replace reduction init axis with new axis if bound to a thread.

### PR DESCRIPTION
**Pull request by csarofeen**
Fixes https://github.com/dmlc/tvm/issues/3407

### CODE REVIEW

**Comment by tqchen** [comment]
Thanks for the PR, sorry for the delayed review as many of folks  I know in the community are traveling to FCRC recently. Can you please add a regression test case to prevent it from happening again?

**Comment by csarofeen** [comment]
Sure, I should be able to base it off the repro in the issue. I'll put it on my todo list but may not get to it for a while. I've been prototyping some kernels in TE, which is why the few small bug fixes. When I get a break I'll come back and write some tests.

**Comment by tqchen** [comment]
@csarofeen can you add a test case?

**Comment by tqchen** [comment]
ping @csarofeen

**Comment by csarofeen** [comment]
Sorry this was so delayed, was actually preparing a talk on TE for CUDA code generation. Simple test added @tqchen

**Comment by csarofeen** [comment]
I can't tell if the linter error is from my change or not. I tried to get it running locally but I got a docker build error.

File: tests/python/unittest/test_codegen_cuda.py [hunk 1]
```diff
@@ -154,9 +154,25 @@ def check_inf_nan(ctx, n, value, dtype):
...
+    B = tvm.compute( (96,), lambda m:
+                     tvm.sum(A[m, k], axis=k),
+                     name='B')
+    s = tvm.create_schedule(B.op)
+
+    s[B].reorder(B.op.reduce_axis[0], B.op.axis[0])
+
+    mo, _ = s[B].split(B.op.axis[0], 32)
+    s[B].bind(mo, tvm.thread_axis("blockIdx.x"))
+    fcuda = tvm.build(s, [A, B], "cuda")
```
Comment (merrymercy) [inline tests/python/unittest/test_codegen_cuda.py]: Add guard to this test (see Line 143). Because some test environments do not have a gpu.

Comment (csarofeen) [reply tests/python/unittest/test_codegen_cuda.py]: Thanks, updated.

**Comment by csarofeen** [comment]
@merrymercy @tqchen I rebased at top of tree and added 2 tests to cover: https://github.com/dmlc/tvm/pull/3444 and https://github.com/dmlc/tvm/pull/3382 hope that's okay.

**Comment by csarofeen** [comment]
Ping. If this looks good I can rebase so it can be merged.

**Comment by tqchen** [comment]
Thanks @csarofeen please rebase

**Comment by csarofeen** [comment]
Looks like this got stuck, is there a way to retry the tests?

**Comment by tqchen** [comment]
Thanks @csarofeen , this PR is now merged!

**Pull request merged**